lib/src/test/kotlin/net/dean/jraw/test/integration/SubmissionReferenceTest.kt

```diff
@@ -194,4 +194,33 @@ class SubmissionReferenceTest : Spek({
             SharedObjects.submittedSelfPost!!.sendReplies(true)
         }
     }
+
+    describe("set spoiler") {
+        assume({ SharedObjects.submittedSelfPost != null }, description = "should have a self post created") {}
+        it("should set/unset the post spoiler flair") {
+            SharedObjects.submittedSelfPost!!.setPostSpoiler(true)
```

Comment:
`SharedObject.submittedSelfPost` is not guaranteed to be non-null here. All `assume` does is execute the code if the condition specified as the first parameter is true. See [here](https://github.com/mattbdean/JRAW/blob/dbbe31999f9403712d236e5e87e515e7479f251f/lib/src/test/kotlin/net/dean/jraw/test/util.kt#L102) for the source.

All you have to do is move the body of the `it` block to the body of the `assume` block.

Reply:
Ah okay, I missed that in my understanding of Spek. Thanks so much! I'll make a follow-up PR with the changes.